Source/Core/DolphinQt2/QtUtils/AspectRatioWidget.cpp

```diff
@@ -0,0 +1,45 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+// Based on:
+// https://stackoverflow.com/questions/30005540/keeping-the-aspect-ratio-of-a-sub-classed-qwidget-during-resize
+
+#include "DolphinQt2/QtUtils/AspectRatioWidget.h"
+
+#include <QBoxLayout>
+#include <QResizeEvent>
+
+AspectRatioWidget::AspectRatioWidget(QWidget* widget, float width, float height, QWidget* parent)
+    : QWidget(parent), m_ar_width(width), m_ar_height(height)
+{
+  m_layout = new QBoxLayout(QBoxLayout::LeftToRight, this);
+
+  // add spacer, then your widget, then spacer
+  m_layout->addItem(new QSpacerItem(0, 0));
+  m_layout->addWidget(widget);
+  m_layout->addItem(new QSpacerItem(0, 0));
+}
+
+void AspectRatioWidget::resizeEvent(QResizeEvent* event)
+{
+  float this_aspect_ratio = (float)event->size().width() / event->size().height();
```

Comment:
Maybe rename this? I find the ``this``-part a bit confusing to me. Maybe ``aspect_ratio`` or ``current_aspect_ratio`` would be preferable.

Reply:
A proper C++ cast would be nice here

Reply:
done